larky/src/main/resources/vendor/ISO8583Decoder.star

```diff
@@ -0,0 +1,505 @@
+        See :mod:`iso8583.specs` module for examples.
+
+    Returns
+    -------
+    int
+        Index in ISO8583 byte array where parsing of message type ended
+
+    Raises
+    ------
+    DecodeError
+        An error decoding ISO8583 bytearray.
+    """
+
+    # Message type is a set length in ISO8583
+    if spec["t"]["data_enc"] == "b":
+        f_len = 2
+    else:
+        f_len = 4
+
+    doc_dec["t"] = ""
+    doc_enc["t"] = {"len": bytes(r"", encoding='utf-8'), "data": bytes(s[idx : idx + f_len])}
+
+    if len(s[idx : idx + f_len]) != f_len:
+        fail(" DecodeError(\n            f\"Field data is {len(s[idx:idx + f_len])} bytes, expecting {f_len}\",\n            s,\n            doc_dec,\n            doc_enc,\n            idx,\n            \"t\",\n        )")
+
+    if spec["t"]["data_enc"] == "b":
+        doc_dec["t"] = s[idx : idx + f_len].hex().upper()
+    else:
+        doc_dec["t"] = s[idx : idx + f_len].decode(spec["t"]["data_enc"])
+
+
+    return idx + f_len
+
+
+def _decode_bitmaps(
+    s,
+    doc_dec,
+    doc_enc,
+    idx,
+    spec,
+    fields,
+) :
+    r"""Decode ISO8583 primary and secondary bitmaps.
+
+    Parameters
+    ----------
+    s : bytes or bytearray
+        Encoded ISO8583 data
+    doc_dec : dict
+        Dict containing decoded ISO8583 data
+    doc_enc : dict
+        Dict containing encoded ISO8583 data
+    idx : int
+        Current index in ISO8583 byte array
+    spec : dict
+        A Python dict defining ISO8583 specification.
+        See :mod:`iso8583.specs` module for examples.
+    fields: set
+        Will be populated with enabled field numbers
+
+    Returns
+    -------
+    int
+        Index in ISO8583 byte array where parsing of bitmaps ended
+
+    Raises
+    ------
+    DecodeError
+        An error decoding ISO8583 bytearray.
+    """
+
+    # Primary bitmap is a set length in ISO8583
+    if spec["p"]["data_enc"] == "b":
+        f_len = 8
+    else:
+        f_len = 16
+
+    doc_dec["p"] = ""
+    doc_enc["p"] = {"len": bytes(r"", encoding='utf-8'), "data": bytes(s[idx : idx + f_len])}
+
+    if len(s[idx : idx + f_len]) != f_len:
+        fail(" DecodeError(\n            f\"Field data is {len(s[idx:idx + f_len])} bytes, expecting {f_len}\",\n            s,\n            doc_dec,\n            doc_enc,\n            idx,\n            \"p\",\n        )")
+
+    if spec["p"]["data_enc"] == "b":
+        doc_dec["p"] = s[idx : idx + f_len].hex().upper()
+        bm = s[idx : idx + f_len]
+    else:
+        doc_dec["p"] = s[idx : idx + f_len].decode(spec["p"]["data_enc"])
+        bm = bytes.fromhex(doc_dec["p"])
+
+
+    fields = sets.union(
+        fields,
+        [
+            byte_idx * 8 + bit
+            for bit in range(1, 9)
+            for byte_idx, byte in enumerate(bm)
+            if byte >> (8 - bit) & 1
+        ]
+    )
+
+    idx += f_len
+
+    # No need to produce secondary bitmap if it's not required
+    if 1 not in fields:
```

Comment:
`not in` will not work, you will need `sets.contains`: 

https://github.com/verygoodsecurity/starlarky/blob/fbf0abc8b0025a4753a5a4a3caeb293f1657f2e4/larky/src/test/resources/stdlib_tests/test_sets.star#L162-L168